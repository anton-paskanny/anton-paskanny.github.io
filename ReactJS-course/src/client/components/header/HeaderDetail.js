import React from 'react';
import { Link } from 'react-router-dom';

import MovieDetail from '../movie/movie-detail/MovieDetail';

import styles from './styles.css';

const HeaderDetail = (props) => (
    <header className="main-header">
        <div className="main-header__inner-wrapper">
            <div className="main-header__top-bar">
                <h1 className="main-header__title common-title">netflixroulette</h1>
                <Link to="/" className="main-header__back-link">Search</Link>
            </div>
            <MovieDetail id={props.match.params.id} />
        </div>
    </header>
);

export default HeaderDetail;