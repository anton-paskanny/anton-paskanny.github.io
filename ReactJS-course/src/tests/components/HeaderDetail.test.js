import React from 'react';
import { shallow } from 'enzyme';

import HeaderDetail from '../../client/components/Header/HeaderDetail';

const MATCH_ID = 123;

let component;
let match;

describe('HeaderDetail', () => {
    beforeEach(() => {
        match = {params: { id: MATCH_ID } };
        component = shallow(<HeaderDetail match={match} />);
    });
    it('render component correctly', () => {
        expect(component).toMatchSnapshot();
    });
    it('should render MovieDetail with passed id in params', () => { 
        expect(component.find('.main-header__back-link').prop('to')).toBe('/'); 
    });
    it('should render link to main page', () => {
        expect(component.find('Connect(MovieDetail)').props().id).toBe(MATCH_ID);
    });
});



