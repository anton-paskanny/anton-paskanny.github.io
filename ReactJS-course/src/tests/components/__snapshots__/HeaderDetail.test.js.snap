// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderDetail render component correctly 1`] = `
<header
  className="main-header"
>
  <div
    className="main-header__inner-wrapper"
  >
    <div
      className="main-header__top-bar"
    >
      <h1
        className="main-header__title common-title"
      >
        netflixroulette
      </h1>
      <Link
        className="main-header__back-link"
        replace={false}
        to="/"
      >
        Search
      </Link>
    </div>
    <Connect(MovieDetail)
      id={123}
    />
  </div>
</header>
`;
